api/metrics/legacy/http.go

```diff
@@ -71,6 +71,14 @@ func WithUIMiddleware(m func(http.Handler) http.Handler) HandlerOption {
 	}
 }
 
+// WithGlobalMiddleware adds a middleware for all operations.
+func WithGlobalMiddleware(m ...func(http.Handler) http.Handler) HandlerOption {
+	return func(h *handlerConfiguration) {
+		h.uiMiddlewares = append(h.uiMiddlewares, m...)
+		h.queryMiddlewares = append(h.queryMiddlewares, m...)
+	}
```

Comment:
do we need to update the legacy path?

Reply:
The legacy path doesn't seem to accept remote writes, so we don't need to update it. Serving just UI and query it's "safe".